pytac/units.py

```diff
@@ -128,31 +167,63 @@ def phys_to_eng(self, value):
         Returns:
             float: The result value.
         """
-        x = self._pre_phys_to_eng(value)
-        result = self._raw_phys_to_eng(x)
-        return result
+        adjusted_value = self._pre_phys_to_eng(value)
+        results = self._raw_phys_to_eng(adjusted_value)
+
+        if self.lower_limit is not None:
+            results = [r for r in results if r >= self.lower_limit]
+        if self.upper_limit is not None:
+            results = [r for r in results if r <= self.upper_limit]
+        if len(results) == 1:
+            return results[0]
+        elif len(results) == 0:
+            raise UnitsException("{0}: Result of conversion "
+                                 "({1}) outside conversion limits ({2}, "
+                                 "{3}).".format(self, results,
+                                                self.lower_limit,
+                                                self.upper_limit))
+        else:
+            raise UnitsException("{0}: There are multiple "
+                                 "corresponding engineering values ({1})."
+                                 .format(self, results))
 
     def convert(self, value, origin, target):
-        """
+        """Convert between two different unit types and chek the validity of
+        the result.
+
         Args:
-            value (float):
+            value (float): the value to be converted
             origin (str): pytac.ENG or pytac.PHYS
             target (str): pytac.ENG or pytac.PHYS
 
         Returns:
-            float: The result value.
+            float: The resulting value.
 
         Raises:
-            UnitsException: invalid conversion.
+            UnitsException: If the conversion is invalid; i.e. if there are no
+                             solutions, or multiple, within conversion limits.
```

Comment:
This now only raises if the unit types are not understood (invalid)

Reply:
Well, no, we'll still get the exception from the other methods we called.